### PR TITLE
make api levels configuratable

### PR DESCRIPTION
**Pull request by deads2k**
Adds options for choosing API levels to start.  This is the first step in disabling "v1beta1".  It also adds warnings to some of our validation methods.

@liggitt

### CODE REVIEW

File: pkg/cmd/server/api/validation/master.go [hunk 1]
```diff
@@ -16,26 +16,60 @@ import (
...
 )
 
-func ValidateMasterConfig(config *api.MasterConfig) fielderrors.ValidationErrorList {
-	allErrs := fielderrors.ValidationErrorList{}
+type ValidationResults struct {
+	Warnings fielderrors.ValidationErrorList
+	Errors   fielderrors.ValidationErrorList
+}
+
+func (r *ValidationResults) AppendErrors(warnings fielderrors.ValidationErrorList, errors fielderrors.ValidationErrorList) {
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: why does `AppendErrors` take `warnings` and `errors` and append both? not sure why this is necessary

File: pkg/cmd/server/api/validation/master.go [hunk 2]
```diff
@@ -16,26 +16,60 @@ import (
...
+	r.AddErrors(errors...)
+	r.AddWarnings(warnings...)
+}
+
+func (r *ValidationResults) Append(additionalResults ValidationResults) {
+	r.AppendErrors(additionalResults.Warnings, additionalResults.Errors)
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: just do this?
```
AddErrors(additionalResults.Errors...)
AddWarnings(additionalResults.Warnings...)
```

File: pkg/cmd/server/api/validation/master.go [hunk 3]
```diff
@@ -47,60 +81,83 @@ func ValidateMasterConfig(config *api.MasterConfig) fielderrors.ValidationErrorL
...
-	allErrs = append(allErrs, ValidateProjectConfig(config.ProjectConfig).Prefix("projectConfig")...)
+	validationResults.AddErrors(ValidateProjectConfig(config.ProjectConfig).Prefix("projectConfig")...)
 
-	return allErrs
+	validationResults.Append(ValidateOpenShiftAPILevels(config.APILevels, api.KnownOpenShiftAPILevels, api.DeadOpenShiftAPILevels, "apiLevels"))
+
+	return validationResults
+}
+
+func ValidateOpenShiftAPILevels(apiLevels []string, knownAPILevels, deadAPILevels []string, name string) ValidationResults {
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: `ValidateAPILevels`

File: pkg/cmd/server/api/validation/master.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/api/validation/master.go]: cruft from other factorizations.

**Comment by liggitt** [comment]
nits, I like it

**Comment by deads2k** [comment]
fixed.

**Comment by deads2k** [comment]
> nits, I like it

enough to give me an lgtm?

**Comment by liggitt** [comment]
yeah, LGTM

**Comment by liggitt** [comment]
weird that gofmt didn't complain about the tabs in the test file originally

**Comment by liggitt** [comment]
[merge]

**Comment by deads2k** [comment]
> weird that gofmt didn't complain about the tabs in the test file originally

it seems that whatever I've got in sublime and used in my local `hack/verify-gofmt.sh` is different than gofmt.  awesome...  re[merge]

**Pull request merged**